src/rubicon/objc/api.py

```diff
@@ -91,9 +91,47 @@
 # the Python objects are not destroyed if they are otherwise no Python references left.
 _keep_alive_objects = {}
 
-# Methods that return an object with is implicitly retained by the caller.
-# See https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/MemoryMgmt/Articles/mmRules.html
-_OWNERSHIP_METHOD_PREFIXES = (b"alloc", b"new", b"copy", b"mutableCopy")
+# Methods that return an object which is implicitly retained by the caller.
+# See https://clang.llvm.org/docs/AutomaticReferenceCounting.html#semantics-of-method-families.
+_RETURNS_RETAINED_FAMILIES = {"init", "alloc", "new", "copy", "mutableCopy"}
+
+
+def get_method_family(method_name: str) -> str:
+    """Returns the method family from the method name. See
+    https://clang.llvm.org/docs/AutomaticReferenceCounting.html#method-families for
+    documentation on method families and corresponding selector names."""
+    method_name = method_name.lstrip("_").split(":")[0]
+    leading_lowercases = []
+    for c in method_name:
+        if c.isupper():
+            break
+        leading_lowercases.append(c)
+    return "".join(leading_lowercases)
+
+
+def method_name_to_tuple(name: str) -> (str, tuple[str, ...]):
+    """
+    Performs the following transformation:
+
+    "methodWithArg0:withArg1:withArg2" -> "methodWithArg0", ("", "withArg1", "withArg2")
```

Comment:
Do I understand correctly that if a selector contains any colons, then it must end with a colon?

Reply:
I believe so, yes, because only the part before the first colon is the message name, and everything that follows must be a named parameter `argumentName:` or an unnamed parameter `:`. The first argument is always unnamed. See https://developer.apple.com/library/archive/documentation/Cocoa/Conceptual/ObjectiveC/Chapters/ocObjectsClasses.html#//apple_ref/doc/uid/TP30001163-CH11-SW6.